### PR TITLE
Make producer work with all consumers

### PR DESCRIPTION
**Pull request by maksimf**

### CODE REVIEW

File: lib/cache/repositories/impressions/redis_repository.rb [hunk 1]
```diff
@@ -39,14 +39,15 @@ def add_bulk(key, bucketing_key, treatments_labels_change_numbers, time)
           # delete fetched impressions afterwards
           def clear
             impressions = impression_keys.each_with_object([]) do |key, memo|
-              _, _, ip, = key.split('/')
+              _, _, ip, _ = key.split('/')
```
Comment (patricioe) [inline lib/cache/repositories/impressions/redis_repository.rb]: too much magic and not very readable.

Mind adding a comment indicating a sample of the input so it's easier for reviews to see what's going on?

Comment (maksimf) [inline lib/cache/repositories/impressions/redis_repository.rb]: yep, of course, moreover it's incorrect here

Comment (maksimf) [inline lib/cache/repositories/impressions/redis_repository.rb]: @patricioe is it ok if I change that for a more readable option?

```ruby
ip = key.split('/')[-2]
```

File: lib/cache/repositories/impressions/redis_repository.rb [hunk 2]
```diff
@@ -39,7 +39,7 @@ def add_bulk(key, bucketing_key, treatments_labels_change_numbers, time)
           # delete fetched impressions afterwards
           def clear
             impressions = impression_keys.each_with_object([]) do |key, memo|
-              _, _, ip, _ = key.split('/')
+              ip = key.split('/')[-2]
```
Comment (patricioe) [inline lib/cache/repositories/impressions/redis_repository.rb]: add still a comment of a sample string you are parsing for the reviewers and for the posterity.

Comment (maksimf) [inline lib/cache/repositories/impressions/redis_repository.rb]: @patricioe done

File: lib/cache/repositories/impressions/redis_repository.rb [hunk 3]
```diff
@@ -39,14 +39,15 @@ def add_bulk(key, bucketing_key, treatments_labels_change_numbers, time)
           # delete fetched impressions afterwards
           def clear
             impressions = impression_keys.each_with_object([]) do |key, memo|
-              _, _, ip, = key.split('/')
+              ip = key.split('/')[-2] # 'prefix/sdk_lang/ip/impressions.name' -> ip
```
Comment (patricioe) [inline lib/cache/repositories/impressions/redis_repository.rb]: what happens if there is an exception parsing? can we catch, continue with next key, clear the cache anyway and perhaps log a single line per batch?

Comment (maksimf) [inline lib/cache/repositories/impressions/redis_repository.rb]: Done

File: lib/cache/repositories/impressions/redis_repository.rb [hunk 4]
```diff
@@ -39,14 +38,19 @@ def add_bulk(key, bucketing_key, treatments_labels_change_numbers, time)
...
+              if ip.nil?
+                @config.logger.warn("Impressions IP parse error for key: #{key}")
```
Comment (patricioe) [inline lib/cache/repositories/impressions/redis_repository.rb]: will the key still be deleted?

File: lib/cache/repositories/impressions/redis_repository.rb (hunk 3 above)
Comment (patricioe) [inline lib/cache/repositories/impressions/redis_repository.rb]: what does -1 do here?

Comment (maksimf) [inline lib/cache/repositories/impressions/redis_repository.rb]: `[-2]` asks for the second element, but it count from right to left
`a = [1, 2, 3]`
`a[-2] # => 2`

File: lib/cache/repositories/impressions/redis_repository.rb (hunk 4 above)
Comment (maksimf) [inline lib/cache/repositories/impressions/redis_repository.rb]: yes if `ip` would be nil I call `next`, i.e. next iteration

**Pull request merged**